include/Event.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "Packet.hpp"
+
+#include <memory>
+#include <cstdint>
+#include <functional>
+
+class Event {
+public:
+    Event(std::uint64_t time, const std::function<void()>& handler);
+
+    std::uint64_t GetTime() const;
+    void Execute() const;
+
+    std::uint64_t execution_time;
```

Comment:
We either need a public getter `GetTime` and private `execution_time` OR public `execution_time`, but not both: if you use `GetTime` then  `execution_time` should be private.